### PR TITLE
Add format options

### PR DESCRIPTION
**Pull request by VitaliyR**
Add format options object which is passed to sharp processor as discussed in #12 

p.s. travis is green for everything except latest node, it can't install sharp there somewhy.

### CODE REVIEW

File: test/index.js [hunk 1]
```diff
@@ -194,4 +195,39 @@ test('defaultComputeFileName', (t) => {
...
+	const scale1 = {
+		format: 'webp',
+		maxWidth: 100,
+		formatOptions: {
+			quality: 80
+		},
+		maxWidth: Number.MAX_SAFE_INTEGER
+	}
+	const defaultFileSize = yield new Promise((res) => resize(file1, scale1, (err, newFile1) => {
+		t.ifError(err)
```
Comment (derhuerst) [inline test/index.js]: Use `reject` from `new Promise((resolve, reject) => {...})` here. The whole test will abort then.

File: test/index.js [hunk 2]
```diff
@@ -194,4 +195,39 @@ test('defaultComputeFileName', (t) => {
...
+		format: 'webp',
+		maxWidth: 100,
+		formatOptions: {
+			quality: 100
+		},
+		maxWidth: Number.MAX_SAFE_INTEGER
+	}
+
+	resize(file2, scale2, (err, newFile2) => {
+		t.ifError(err)
```
Comment (derhuerst) [inline test/index.js]: As above.

File: test/index.js [hunk 3]
```diff
@@ -194,4 +195,39 @@ test('defaultComputeFileName', (t) => {
...
+		t.ok(newFile2[SHARP_INFO].size > defaultFileSize)
+	});
+
+	t.end()
```
Comment (derhuerst) [inline test/index.js:236]: Please move this to the below the `.size > defailtFileSize` check inside the callback.

**Comment by derhuerst** [comment]
FYI: The CI is failing for a different reason.

File: test/index.js (hunk 1 above)
Comment (VitaliyR) [reply test/index.js]: Didn't get it. Should I just do:
```js
err && rej(err)
```
?

File: test/index.js (hunk 3 above)
Comment (VitaliyR) [reply test/index.js:236]: Whooops, that was silly, thanks

**Comment by VitaliyR** [comment]
@derhuerst I dont see anything related to this plugin in npm logs. Its something about building sharp from C files, as far as I'm getting it

File: test/index.js (hunk 1 above)
Comment (derhuerst) [reply test/index.js]: Yes, put please like `if (err) return reject(err)` for readability.

**Comment by VitaliyR** [comment]
@derhuerst done. Also I've refactored 2nd Promise

File: test/index.js (hunk 3 above)
Comment (derhuerst) [reply test/index.js:236]: ping 😉

File: test/index.js [hunk 4]
```diff
@@ -194,4 +195,40 @@ test('defaultComputeFileName', (t) => {
...
+	const scale1 = {
+		format: 'webp',
+		maxWidth: 100,
+		formatOptions: {
+			quality: 80
+		},
+		maxWidth: Number.MAX_SAFE_INTEGER
+	}
+	const defaultFileSize = yield new Promise((res, rej) => resize(file1, scale1, (err, newFile1) => {
+		if (err) rej(err);
```
Comment (derhuerst) [inline test/index.js]: `return` here or use `else `res(…)`.

File: test/index.js (hunk 3 above)
Comment (VitaliyR) [reply test/index.js:236]: I've put there yield so it will wait until resolved. I thought it would be better for reading.

**Pull request merged**